recipes/openvdb/meta.yaml

```
@@ -0,0 +1,53 @@
+{% set name = "openvdb" %}
+{% set version = "10.0.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/AcademySoftwareFoundation/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 887a3391fbd96b20c77914f4fb3ab4b33d26e5fc479aa036d395def5523c622f
+
+
+build:
+  number: 0
+  skip: true  # [win or osx]
+
```

```suggestion
build:
  number: 0
  skip: true  # [win or osx]
  run_exports:
    - {{ pin_subpackage('openvdb', max_pin='x.x.x') }}
```
Run exports are required for packages that ship libraries which can be dynamically linked by other packages. Adjust the max_pin for the ABI compatability guarantees of your package.

---

openvdb guarantees ABI compatability between version in the same major release, see [these references to `OPENVDB_ABI_VERSION_NUMBER`](https://github.com/search?q=repo%3AAcademySoftwareFoundation%2Fopenvdb%20OPENVDB_ABI_VERSION_NUMBER&type=code).

Meaning that the line should be:
```
    - {{ pin_subpackage('openvdb', max_pin='x') }}
 ```

---

Is the ABI version separate from the API version? In that case, we can easily make a subpackage like `libvdb10` which would contain `libvdb.so.10`. That way the API and ABI can be depended on separately.

---

Major is incremented on ABI changes for the Grid class and its constituent classes or changes to file format. Minor on API. I think your suggestion would make a lot of sense. I'll have to educate myself a bit more on subpackages in the conda-forge ecosystem and then test this out :).

---

https://www.openvdb.org/documentation/doxygen/faq.html#sVersionNumbering

---

I looked into this where I used one output named libopenvdb, and added `    - {{ pin_subpackage('openvdb', max_pin='x') }}`  to the top run requires. (I also tested this with two outputs, libopenvdb and openvdb..). Didn't really get this to work. I have been trying to find more documentation on split packages / metapackages, but haven't really grokked it yet. I've also tried to find some other feedstock-recipes that does something similar but the ones I find are quite complicated.

Here are some more direct questions:

* Should there be one or two subpackages under `outputs:` ? If one, should I explicitly exclude the shared lib from the top level package?
* Should the library only package be named libopenvdb or libopenvdb10?
* Should I have no build steps in top leve / openvdb subpackage and only build in libopenvdb?

---

Just to reiterate, creating a `libopenvdb10` package isn't really needed unless the ABI and API versions are separate. But it looks like [you are treating them as the same](https://github.com/versatran01/staged-recipes/blob/12481bcf36e1c4f0cc9d285d0b1382f49a58ecfc/recipes/openvdb/meta.yaml#L34C41-L34C41).

For multiple outputs, think of two steps:

1. Build all the artifacts and install them to $PREFIX. The top level package should be called `openvdb-split`, and it does this. Just build everything here together.
2. Separate the artifacts from `openvdb-split` into multiple outputs. In the output section you list all of the package that will be published. Use the glob matching file lists to separate the files from $PREFIX into each output.

Probably, the outputs for this recipe are `openvdb` and `libopenvdb10`

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#specifying-files-to-include-in-output

Here's an live example feedstock, but they use install scripts instead of file lists to separate the outputs. IMHO, file lists are more verbose, but simpler and easier to understand.

https://github.com/conda-forge/mlir-feedstock/blob/main/recipe/meta.yaml

---

Thanks! I'll do some more testing on this. And I would be interested in hearing if @versatran01 has any thoughts or opinions. 

It can always be a later enhancement, right? This would be better than no openvdb on conda-forge.

---

Please add an appropriate run_export. This is required so that downstream packages have a compatible version of this library at runtime.